src/scipp/curve_fit.py

```diff
@@ -267,6 +325,45 @@ def _curve_fit(
     dgcov[:] = pcov
 
 
+def _curve_fit_chunk(
+    coords,
+    f,
+    da,
+    p0,
+    bounds,
+    map_over,
+    unsafe_numpy_f,
+    kwargs,
+):
```

Comment:
Is it really useful to add type hints to helper functions? I typically don't do that.

Or do you mean it would be good to have for this particular function but not for the other helper functions in this module?

Reply:
Depends. It can help the reader (reviewer) a lot.